backend/dataall/aws/handlers/sts.py

```diff
@@ -173,9 +155,18 @@ def get_delegation_role_arn(cls, accountid):
         Returns:
                 string : arn of the delegation role on the target aws account id
         """
-        return 'arn:aws:iam::{}:role/{}'.format(
-            accountid, cls.get_delegation_role_name()
-        )
+        return 'arn:aws:iam::{}:role/{}'.format(accountid, cls.get_delegation_role_name())
+
+    @classmethod
+    def get_cdk_look_up_role_arn(cls, accountid, region):
+        """Returns the name that will be assumed to perform IAM actions on a given AWS accountid using CDK Toolkit role
+        Args:
+            accountid(string) : aws account id
+        Returns:
+                string : arn of the CDKToolkit role on the target aws account id
+        """
+        log.info(f"Getting CDK look up role: arn:aws:iam::{accountid}:role/cdk-hnb659fds-lookup-role-{accountid}-{region}")
```

Comment:
Yes, it looks weird. It comes from the CDK [bootstrapping ](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html)command, it is a qualifier that is used to avoid naming conflicts. The default value is `hnb659fds`, so if customers run the command as we define it in the frontend they would get the default. If they have any other kind of bootstrapping standard we would run into issues. I think there is already an issue about custom bootstrapping processes.